api/app/utils.py

```diff
@@ -38,3 +40,29 @@ def filter(self, record):
             self.warning_count += 1
 
         return False
+
+
+def make_request_with_retries(
```

Comment:
Do we need something specific, or could we rely on the Retry util directly?
from requests.adapters import HTTPAdapter, Retry

Reply:
The retry utility has some useful functionality that this approach didn't have (back-off timing and status filtering for retries specifically). I've updated this function to wrap the retries utility. This `make_request_with_retries` function is still useful to reduce redundancy